src/cmem/ActivityControl/SilkActivityControl.tsx

```diff
@@ -251,59 +247,61 @@ export function SilkActivityControl({
     const waitingProgress = activityStatus && activityStatus.concreteStatus === "Waiting";
     const animateProgress = activityStatus && activityStatus.progress > 0 && activityStatus.progress < progressBreakpointAnimation;
     const indeterminateProgress = activityStatus && activityStatus.progress < progressBreakpointIndetermination;
+    const intent = activityStatus ? calcIntent(activityStatus) : "none";
 
     if (visualization === "progressbar") {
         visualizationProps = {
             progressBar: {
                 animate: waitingProgress || (runningProgress && animateProgress),
                 stripes: waitingProgress || (runningProgress && animateProgress),
                 value: (waitingProgress || (runningProgress && indeterminateProgress)) ? undefined : ((activityStatus && activityStatus.progress > 0) ? (activityStatus.progress / 100) : 0),
-                intent: activityStatus ? calcIntent(activityStatus) : "none",
+                intent
             }
         }
     };
     if (visualization === "spinner") {
         visualizationProps = {
             progressSpinner: {
                 value: (waitingProgress || (runningProgress && indeterminateProgress)) ? undefined : ((activityStatus && activityStatus.progress > 0) ? (activityStatus.progress / 100) : 0),
-                intent: activityStatus ? calcIntent(activityStatus) : "none",
+                intent
             }
         }
     };
-    
-    if(progressFinishedIcon) {
-         visualizationProps = {
-              ...visualizationProps,
-              progressFinishedIcon
-         }
-    }
+ 
 
-    return <>
-        <ActivityControlWidget
-            key={"activity-control"}
-            tags={tags}
-            data-test-id={props["data-test-id"]}
-            label={activityControlLabel}
-            activityActions={actions}
-            statusMessage={activityStatus?.message}
-            {...visualizationProps}
-            {...otherLayoutConfig}
-        />
-        {errorReport && failureReportAction && <ActivityExecutionErrorReportModal
-            title={failureReportAction.title}
-            key={"error-report-modal"}
-            closeButtonValue={failureReportAction.closeButtonValue}
-            downloadButtonValue={failureReportAction.downloadButtonValue}
-            fetchErrorReport={async () => {
-                return await failureReportAction.fetchErrorReport(true) as (string | undefined)
-            }}
-            report={failureReportAction.renderReport(errorReport)}
-            onDiscard={closeErrorReport}
-        />}
-    </>
+    return (
+        <>
+            <ActivityControlWidget
+                key={"activity-control"}
+                tags={tags}
+                data-test-id={props["data-test-id"]}
+                label={activityControlLabel}
+                activityActions={actions}
+                statusMessage={activityStatus?.message}
+                progressSpinnerFinishedIcon={
```

Comment:
This cannot work correctly and will break the usage of `SilkActivityControl`. This implementation would always display the icon in case there is an intent different from `none`. `SilkActivityControl` as a wrapper for `ActivityControlWidget` should probably not decide about the icon usage, this should be done outside. Imho it is better also to let set an icon for `SilkActivityControl` and when set, then route it through.

Reply:
Yeah, thanks I can see that now, an oversight on my path. However, I would prefer the deciding logic that determines showing or not showing the `progressFinishedIcon` to be in the `SilkActivityControl` it would be unnecessarily repetitive to have this in every overarching component that uses the silkActivityControl.

Also, the way the `ActivityList` component is wired only the children components get rendered for every change (webSocket update) as opposed to the parent component it self. So, getting the current status and deciding whether or not to show the parent component might be a bit hard

Reply:
Ok, in case we have have a very clear finished state for activities, logic may also work in `SilkActivityControl`, at least we can try it out how far we come.